service/src/test/java/bio/terra/workspace/service/resource/controlled/flight/clone/workspace/CloneCreateCloudContextFlightTest.java

```diff
@@ -0,0 +1,131 @@
+import bio.terra.workspace.service.workspace.GcpCloudContextService;
+import bio.terra.workspace.service.workspace.model.*;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mock;
+
+@Tag("unit")
+public class CloneCreateCloudContextFlightTest extends BaseMockitoStrictStubbingTest {
+  @Mock private FlightBeanBag flightBeanBag;
+  @Mock private GcpCloudContextService gcpCloudContextService;
+  @Mock private AzureCloudContextService azureCloudContextService;
+  @Mock private SpendProfileService spendProfileService;
+  @Mock private GcpCloudContext gcpCloudContext;
+  private final AuthenticatedUserRequest userRequest = new AuthenticatedUserRequest();
+  private final UUID sourceWorkspaceId = UUID.randomUUID();
+  private final SpendProfileId sourceSpendProfileId =
+      new SpendProfileId(UUID.randomUUID().toString());
+  private final AzureCloudContext azureCloudContext =
+      new AzureCloudContext(
+          null,
+          new CloudContextCommonFields(
+              sourceSpendProfileId,
+              WsmResourceState.READY,
+              /* flightId= */ null,
+              /* error= */ null));
+
+  @BeforeEach
+  public void beforeEach() {
+    when(flightBeanBag.getGcpCloudContextService()).thenReturn(gcpCloudContextService);
+    when(flightBeanBag.getAzureCloudContextService()).thenReturn(azureCloudContextService);
+  }
+
+  @Test
+  void testCreateFlightWithNoSourceCloudContext() {
+    assertExpectedSteps(new ArrayList<>(List.of(CreateCloudContextIdsForFutureStepsStep.class)));
+  }
+
+  @Test
+  void testCreateFlightWithSourceGCPCloudContext() {
+    when(gcpCloudContextService.getGcpCloudContext(sourceWorkspaceId))
+        .thenReturn(Optional.of(gcpCloudContext));
+    assertExpectedSteps(
+        new ArrayList<>(
+            List.of(
+                CreateCloudContextIdsForFutureStepsStep.class,
+                LaunchCreateCloudContextFlightStep.class,
+                AwaitCreateCloudContextFlightStep.class)));
+  }
+
+  @Test
+  void testCreateFlightWithSourceAzureCloudContext() {
+    when(azureCloudContextService.getAzureCloudContext(sourceWorkspaceId))
+        .thenReturn(Optional.of(azureCloudContext));
+    assertExpectedSteps(
+        new ArrayList<>(
+            List.of(
+                CreateCloudContextIdsForFutureStepsStep.class,
+                LaunchCreateCloudContextFlightStep.class,
+                AwaitCreateCloudContextFlightStep.class)));
+  }
+
+  @Test
+  void testUsesSourceSpendProfileIfNoDestinationSpendProfile() {
+    when(flightBeanBag.getSpendProfileService()).thenReturn(spendProfileService);
+    FeatureConfiguration config = new FeatureConfiguration();
+    // This is a misnomer... it is not specific to GCP (and in reality, always on).
```

Comment:
Good call out -- can you drop a ticket to remove this flag? I don't think it's needed anymore.